src/pocketmine/Player.php

```diff
@@ -2941,6 +2942,13 @@ public function handlePlayerAction(PlayerActionPacket $packet) : bool{
 						$this->level->broadcastLevelEvent($pos, LevelEventPacket::EVENT_BLOCK_START_BREAK, (int) (65535 / $breakTime));
 					}
 				}
+				if($target instanceof ItemFrameBlock){
+					$pk = new ItemFrameDropItemPacket();
+					$pk->x = (int) $target->x;
+					$pk->y = (int) $target->y;
+					$pk->z = (int) $target->z;
+					$this->handleItemFrameDropItem($pk);
```

Comment:
This block should be placed higher up, since the breaking animation shouldn't be broadcast for removing items from an item frame. I'm also not sure whether extinguishing fire takes precedence over removing an item from a frame or not.